### PR TITLE
[Prover] Sequencer node stake table initialization

### PR DESCRIPTION
**Pull request by rob-maron**
Allows the prover to pull in the stake table from sequencer nodes running with `--` `config`. 

This makes it so that a transient failure of both the prover and orchestrator does not cause the prover to lose the stake table permanently. This would cause it to stop working entirely

### CODE REVIEW

File: hotshot-state-prover/src/service.rs [hunk 1]
```diff
@@ -103,62 +102,70 @@ pub fn init_stake_table(
     Ok(st)
 }
 
-async fn init_stake_table_from_orchestrator(
-    orchestrator_url: &Url,
+#[derive(Debug, Deserialize)]
+pub struct PublicHotShotConfig {
```
Comment (jbearer) [inline hotshot-state-prover/src/service.rs:106]: Nit: why not just import this from sequencer?

**Review by mrain — APPROVED** [review]
LGTM

**Review by jbearer — APPROVED** [review]
Nice

File: hotshot-state-prover/src/service.rs (hunk 1 above)
Comment (rob-maron) [reply hotshot-state-prover/src/service.rs:106]: I tried that but cargo told me there was a cyclic dependency problem somewhere. I also considered moving it to `utils` but it didn't seem in the util spirit. 

I can keep pushing if we want though, maybe it wouldn't be too bad

Comment (mrain) [reply hotshot-state-prover/src/service.rs:106]: Deploying the light client contract will invoke `light_client_genesis` fn from `hotshot-state-prover` repo. Thus a cyclic dep.

**Pull request merged**